corgidrp/astrom.py

```diff
@@ -649,14 +734,130 @@ def compute_boresight(image, source_info, target_coordinate, cal_properties):
+
+    Args:
+        input_dataset (corgidrp.data.Dataset): corgidrp dataset object with images to compute the distortion from
+        pos1 (np.array): 2D array of the (x, y) pixel positions for the first star in every star pair
+        meas_offset (np.array): 2D array of the (delta_x, delta_y) values for each star from the first star position
+        sky_offset (np.array): 2D array of the (delta_ra, delta_dec) offsets between the matched stars in the reference field
+        meas_errs (np.array): 2D array of the (x_err, y_err) error in the measured pixel positions
+        platescale (float): Platescale value to use in computing distortion
+        northangle (float): Northangle value to use in computing distortion 
+        fitorder (int): The order of legendre polynomial to fit to the image distortion (default: 3)
+        initial_guess (np.array): Initial guess of fitting parameters (legendre coefficients) length based on fitorder (2 * (fitorder+1)**2), (default: None)
+
+    Returns:
+        distortion_coeffs (tuple): The legendre coefficients (np.array) and polynomial order used for the fit (int)
+
+    '''
+
+    ## SET FITTING PARAMS
+    # assume all images in dataset have the same shape
+    input_image = input_dataset[0].data
+    x0 = np.shape(input_image)[1] // 2
+    y0 = np.shape(input_image)[0] // 2
+    
+    # define fitting params            
+    fitparams = (fitorder + 1)**2
+    
+    # initial guesses for the legendre coeffs if none are passed
+    if initial_guess is None:
+        initial_guess = [0 for _ in range(fitorder+1)] + [500,] + [0 for _ in range(fitparams - fitorder - 2)] + [0,500] + [0 for _ in range(fitparams-2)]
+    
+    ## OPTIMIZE 
+    # first_stars_, offsets_, true_offsets_, errs_ = first_stars, offsets, true_offsets, errs
+    (distortion_coeffs, _) = optimize.leastsq(fit_distortion_solution, initial_guess, 
+                                              args=(fitorder, platescale, 
+                                                northangle, pos1, meas_offset, 
+                                                sky_offset, meas_errs, x0, y0))
+
+    return (distortion_coeffs, fitorder)
+  
+  
+def boresight_calibration(input_dataset, field_path='JWST_CALFIELD2020.csv', field_matches=None, find_threshold=10, fwhm=7, mask_rad=1, 
+                          comparison_threshold=50, search_rad=0.0075, platescale_guess=21.8, platescale_tol=0.1, center_radius=0.9, 
+                          frames_to_combine=None, find_distortion=False, fitorder=3, position_error=None, initial_dist_guess=None):
     """
     Perform the boresight calibration of a dataset.
     
     Args:
         input_dataset (corgidrp.data.Dataset): Dataset containing a images for astrometric calibration
         field_path (str): Full path to file with search field data (ra, dec, vmag, etc.) (default: 'JWST_CALFIELD2020.csv')
```

Comment:
Is it safe to have this as a missable keyword argument set by default to 'JWST_CALFIELD2020.csv', or would it be better just to have it be another required argument?

Reply:
this file is packaged into corgidrp, so the default should always work.